==== ntpd/ntp_proto.c ====
2010-10-22 01:55:45-04:00, stenn@deacon.udel.edu +2 -5
  [Bug 1670] Fix peer->bias and broadcastdelay

--- 1.307/ntpd/ntp_proto.c	2010-10-11 21:06:05 -07:00
+++ 1.308/ntpd/ntp_proto.c	2010-10-21 22:55:45 -07:00
@@ -929,7 +929,6 @@ receive(
 
 			} else {
 				peer->delay = sys_bdelay;
-				peer->bias = -sys_bdelay / 2.;
 			}
 			break;
 		}
@@ -1570,7 +1569,6 @@ process_packet(
 		p_del = fabs(t21 - t34);
 		p_offset = (t21 + t34) / 2.;
 	}
-	p_offset += peer->bias;
 	p_disp = LOGTOD(sys_precision) + LOGTOD(peer->precision) +
 	    clock_phi * p_del;
 
@@ -1647,7 +1645,7 @@ process_packet(
 	/*
 	 * That was awesome. Now hand off to the clock filter.
 	 */
-	clock_filter(peer, p_offset, p_del, p_disp);
+	clock_filter(peer, p_offset + peer->bias, p_del, p_disp);
 
 	/*
 	 * If we are in broadcast calibrate mode, return to broadcast
